src/Components/Onboarding/ItemLink.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react"
+import styled, { StyledFunction } from "styled-components"
+
+import { fadeIn, fadeOut } from "../../Assets/Animations"
+import * as fonts from "../../Assets/Fonts"
+
+import Icon from "../Icon"
+
+const anchor: StyledFunction<State & React.HTMLProps<HTMLInputElement>> = styled.a
+const Link = anchor`
+  display: flex;
+  font-size: 14px;
+  color: black;
+  text-decoration: none;
+  text-transform: uppercase;
+  font-family: ${fonts.primary.fontFamily};
+  border-top: 1px solid #e5e5e5;
+  &:hover {
+    background-color: #f8f8f8;
+  }
+  ${props => (props.fadeIn ? fadeIn : null)}
+  ${props => (props.fadeOut ? fadeOut : null)}
+`
+
+const Avatar = styled.img`
+  padding: 10px 15px 10px 10px;
+`
+
+const FullWidthCol = styled.div`
+  flex: 1;
+  display: flex;
+  align-items: center;
+`
+
+const Col = styled.div`
+  display: flex;
+  align-items: center;
+`
+
+export interface State {
+  fadeIn: boolean
+  fadeOut: boolean
+  item_id: string
+}
+
+interface Props extends React.HTMLProps<HTMLAnchorElement> {
+  item?: any
+  id: string
+  _id: string
+  name: string
+  image_url
+}
+
+export default class ItemLink extends React.Component<Props, State> {
+  state = {
+    fadeIn: false,
+    fadeOut: false,
+    item_id: this.props.id,
```

Comment:
This doesn’t really work, does it? If your initial state isn’t static, then you should set it from a `constructor` override.

Reply:
No I think this is left over code from @yuki24 first implemented the artist component.

Reply:
I was so sure I did use this syntax at some point (without `this.props.id` for sure) but looking at git commits it seems like I actually used [`constructor`](https://github.com/artsy/reaction/pull/268/files#diff-617784f1b8c8a101065da7cc0270070cR66) instead, which later was re-written to [not use it](https://github.com/artsy/reaction/pull/285/files#diff-17dd94b4bd0e5bcc37ff8e8aad9e2027R49).